aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -2243,6 +2243,74 @@ Tensor nonzero_cpu(const Tensor& self) {
   return result;
 }
 
+Tensor& nonzero_static_out_cpu(
+    const Tensor& self,
+    int64_t size,
+    int64_t fill_value,
+    Tensor& result) {
+  // Sanity check on input
+  TORCH_CHECK(size >= 0, "'size' must be an non-negative integer");
+  int64_t ndim = self.dim();
+  TORCH_CHECK(ndim > 0, "Rank must be positive");
```

Comment:
OK, you know what, I changed my mind, we should allow zero rank as it generalizes cleanly.

Reply:
What is the expected behavior of the zero rank input? Same as regular `nonzero()`?